rag_experiment_accelerator/config/index_config.py

```diff
@@ -132,7 +132,7 @@ def from_index_name(cls, index_name: str) -> "IndexConfig":
         Reverse of index_name().
         """
 
-        key_values = [kv.split("-") for kv in index_name.split("_")]
+        key_values = [kv.split("-", 1) for kv in index_name.split("_")]
         properties = {kv[0]: kv[1].strip() for kv in key_values}
```

Comment:
Severity: Medium - Changing the split method to `split('-', 1)` ensures that only the first occurrence of `-` is used for splitting, which prevents potential issues if there are multiple hyphens in the key-value pairs. This improves the robustness of the parsing logic. However, ensure that the resulting `key_values` list contains exactly two elements to avoid potential `IndexError`.


[Hack - peer-review-pal] We are sorry for the intrusion. Feel free to ignore/delete the comments with the tag - they are automatically generated by [our Hack project](https://github.com/microsoft/peer-review-pal).
If you find them useful, want us to delete them or have any feedback or know how to make it better and want to contribute - reach out to some of us: juliame@microsoft.com vadimkirilin@microsoft.com oteslenko@microsoft.com vimeena@microsoft.com
You can prevent future comments from appearing by adding #prp-ignore to your PR description.